pkg/flowaggregator/flowaggregator.go

```diff
@@ -252,19 +268,62 @@ func (fa *flowAggregator) Run(stopCh <-chan struct{}) {
 		klog.InfoS("Object stores synced")
 	}()
 
+	if fa.certificateProvider != nil {
+		fa.certificateProvider.AddListener(fa)
+
+		wg.Add(1)
+		go func() {
+			defer wg.Done()
+			fa.certificateProvider.Run(stopCh)
+		}()
+
+		if !cache.WaitForNamedCacheSync("flowAggregator", stopCh, fa.certificateProvider.HasSynced) {
+			return
+		}
+
+		if err := fa.InitCollectors(); err != nil {
+			klog.ErrorS(err, "unable to create collectors")
+			return
+		}
```

Comment:
this needs some comments, including an explanation as to why we need a special case (why not wait for `CertificateUpdated` to be called the first time?)